enterprise/frontend/src/embedding-sdk/hooks/private/get-auth-configuration.ts

```diff
@@ -0,0 +1,61 @@
+import { match, P } from "ts-pattern";
+
+import type { EmbeddingSessionToken, SDKConfig } from "embedding-sdk";
+import { presentApiKeyUsageWarning } from "embedding-sdk/lib/user-warnings";
+import { getErrorMessage } from "embedding-sdk/lib/user-warnings/constants";
+import {
+  getOrRefreshSession,
+  setLoginStatus,
+} from "embedding-sdk/store/reducer";
+import type { SdkDispatch } from "embedding-sdk/store/types";
+import type {
+  SDKConfigWithApiKey,
+  SDKConfigWithJWT,
+} from "embedding-sdk/types";
+import api from "metabase/lib/api";
+import type { Dispatch } from "metabase-types/store";
+
+const setupJwtAuth = (
+  jwtProviderUri: SDKConfigWithJWT["jwtProviderUri"],
+  dispatch: SdkDispatch,
+) => {
+  api.onBeforeRequest = async () => {
+    const tokenState = await dispatch(getOrRefreshSession(jwtProviderUri));
+
+    api.sessionToken = (tokenState.payload as EmbeddingSessionToken | null)?.id;
+  };
+
+  dispatch(setLoginStatus({ status: "validated" }));
+};
+
+const setupLocalApiKey = (
+  dispatch: SdkDispatch,
+  apiKey: SDKConfigWithApiKey["apiKey"],
+) => {
+  api.apiKey = apiKey;
+  dispatch(setLoginStatus({ status: "validated" }));
+};
+
+export const getAuthConfiguration = (
+  config: SDKConfig,
+  dispatch: Dispatch,
+  appName: string,
+) =>
+  match<[SDKConfig, string], string | void>([config, window.location.hostname])
+    .with(
+      [
+        {
+          jwtProviderUri: P.select(P.not(P.nullish).and(P.string.minLength(1))),
+        },
+        P._,
+      ],
+      jwtProviderUri => setupJwtAuth(jwtProviderUri, dispatch),
+    )
+    .with([{ apiKey: P.select(P.not(P.nullish)) }, "localhost"], apiKey => {
```

Comment:
```suggestion
    .with([{ apiKey: P.select(P.nonNullable) }, "localhost"], apiKey => {
```
This might work.

Also, this gets me thinking if the condition should be the same as the `jwtProviderUri` i.e. either check non-null + minlength or just non-null since both are strings.